db/memorystore.go

```diff
@@ -116,6 +119,31 @@ func (m *MemoryStore) AddAccount(acct *core.Account) (int, error) {
 	return len(m.accountsByID), nil
 }
 
+func (m *MemoryStore) ChangeAccountKey(acct *core.Account, newKey *jose.JSONWebKey) (error, *core.Account) {
```

Comment:
Can you swap the order of the `*core.Account` and `error` returns? It's common Go convention to always return the `error` type as the last return value.

Reply:
Re-reading this I understand now that the `*core.Account` return was being used for key-conflicts. I think you should remove this return entirely and just return an `error` and use the `existingAccountError` I describe in my other comment for that case.